src/routes/jira/server/jira-server-url-post.ts

```diff
@@ -3,34 +3,88 @@ import axios from "axios";
 import { GitHubServerApp } from "models/github-server-app";
 import { isValidUrl } from "utils/is-valid-url";
 
+interface MessageAndCode {
+	errorCode: string;
+	message: string;
+	statusCode: number;
+}
+
+interface GheServerUrlErrorResponses {
+	[key: string | number]: MessageAndCode;
+}
+
+interface GheServerUrlErrors {
+	codeOrStatus: GheServerUrlErrorResponses;
+}
+
+const gheServerUrlErrors: GheServerUrlErrors = {
+	codeOrStatus: {
+		invalidUrl: {
+			errorCode: "GHE_ERROR_INVALID_URL",
+			message: "Invalid URL",
+			statusCode: 200
+		},
+		ENOTFOUND: {
+			errorCode: "GHE_ERROR_ENOTFOUND",
+			message: "Request to URL failed",
+			statusCode: 200
+		},
+		502: {
+			errorCode: "GHE_SERVER_BAD_GATEWAY",
+			message: "Bad gateway",
+			statusCode: 502
+		},
+		default: {
+			errorCode: "GHE_ERROR_DEFAULT",
+			message: "Something went wrong",
+			statusCode: 200
+		}
+	}
+};
+
+export const getGheErrorMessages = (codeOrStatus: number | string | null) => {
```

Comment:
not sure this function is needed, couldn't you just do `const errorMessage = gheServerUrlErrors[codeOrStatus] || gheServerUrlErrors.default;` in the try-catch below?

Reply:
I was doing that initially and then added the switch. But you're right, it's a little redundant.